### PR TITLE
Allow instance member accesses in top-level field initializers

### LINKED ISSUE #5069: Completion doesn't include locals declared in previous submission
Try typing the code below and notice that the completion list that pops up when the highlighted text below is typed does not contain the variable “number” which was declared in the previous line.
```
> int number = 10;
> int n1 = num          
```

Was internal issue 531487

### PR DESCRIPTION
**Pull request by amcasey**
...as if they were locals.

Fixes #5069

### CODE REVIEW

**Comment by amcasey** [comment]
FYI @rchande @Pilchie @DustinCampbell @jasonmalinowski @CyrusNajmabadi  for completion
FYI @cston @ManishJayaswal @KevinH-MS @tmat  for interactive

File: src/EditorFeatures/CSharpTest/Completion/CompletionProviders/SymbolCompletionProviderTests.cs [hunk 1]
```diff
@@ -8304,6 +8304,31 @@ public void StaticMembersInPropertyInitializer()
...
+class A {
+    Action abc;
+    event Action B = $$
+}
+";
+            VerifyItemIsAbsent(markup, "abc");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.Completion)]
+        public void StaticMembersInFieldLikeEventInitializer()
```
Comment (rchande) [inline src/EditorFeatures/CSharpTest/Completion/CompletionProviders/SymbolCompletionProviderTests.cs]: Renamed wrong method?

**Comment by rchande** [comment]
:+1:

File: src/EditorFeatures/CSharpTest/Completion/CompletionProviders/SymbolCompletionProviderTests.cs (hunk 1 above)
Comment (amcasey) [inline src/EditorFeatures/CSharpTest/Completion/CompletionProviders/SymbolCompletionProviderTests.cs]: Yep.  Thanks!

**Comment by cston** [comment]
LGTM

**Comment by jasonmalinowski** [comment]
:+1:

**Pull request merged**